.github/actions/dev-env-setup/action.yml

```diff
@@ -17,26 +17,23 @@ runs:
         echo "PERL_MB_OPT=--install_base '/home/runner/perl5'" >> $GITHUB_ENV
         echo "PERL_MM_OPT=INSTALL_BASE=/home/runner/perl5" >> $GITHUB_ENV
         echo "/home/runner/perl5/bin" >> $GITHUB_PATH
-    - uses: actions/cache@v2
+    - uses: actions/cache@v3
       id: asdf-cache
       with:
         path: |
           ~/.asdf
-        key: ${{ runner.os }}-asdf-cache-${{ hashFiles('.tool-versions') }}
-    - uses: actions/cache@v2
+        key: ${{ runner.os }}-asdf-cache-${{ hashFiles('.tool-versions') }}-v2
```

Comment:
clobbering the old one didn't work? needed a fresh start?

Reply:
Yes, because if we have a cache hit, GH actions won't save the cache, and we had a cache hit given that the tool-versions wasn't changed

Reply:
maybe we should add a `hashfiles('.github/actions/dev-env-setup/action.yml')` in as well; bumping to cache@v3 should invalidate too